lib/livebook/runtime/evaluator.ex

```diff
@@ -475,17 +475,22 @@ defmodule Livebook.Runtime.Evaluator do
     end
 
     state = put_context(state, ref, new_context)
-
     output = Evaluator.Formatter.format_result(result, language)
 
+    definitions =
+      for {module, _} <- tracer_info.modules_defined,
+          definition = get_definition(context, module),
+          do: definition
```

Comment:
Let's move this to the success clause above, same as `identifiers_defined`. If the evaluation fails, we effectively revert any module definitions.

Also, it's small, but let's move it to a separate function, as in `definitions = definitions(tracer_info)`.